packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -112,127 +51,33 @@ window.\$hotReloadHook = function(modules) {
-      return el[0].getAttribute("href");
-    }
-  }
-  // return default value
-  return "/";
-}());
-$_currentDirectoryScript
-// dart loader
-if(!window.\$dartLoader) {
-   window.\$dartLoader = {
-     appDigests: _currentDirectory + 'basic.digests',
-     moduleIdToUrl: new Map(),
-     urlToModuleId: new Map(),
-     rootDirectories: new Array(),
-     // Used in package:build_runner/src/server/build_updates_client/hot_reload_client.dart
-     moduleParentsGraph: new Map(),
-     moduleLoadingErrorCallbacks: new Map(),
-     forceLoadModule: function (moduleName, callback, onError) {
-       if (typeof onError != 'undefined') {
-         var errorCallbacks = \$dartLoader.moduleLoadingErrorCallbacks;
-         if (!errorCallbacks.has(moduleName)) {
-           errorCallbacks.set(moduleName, new Set());
-         }
-         errorCallbacks.get(moduleName).add(onError);
-       }
-       requirejs.undef(moduleName);
-       requirejs([moduleName], function() {
-         if (typeof onError != 'undefined') {
-           errorCallbacks.get(moduleName).delete(onError);
-         }
-         if (typeof callback != 'undefined') {
-           callback();
-         }
-       });
-     },
-     getModuleLibraries: null, // set up by _initializeTools
-   };
-}
-let modulePaths = {};
-let customModulePaths = {};
-window.\$dartLoader.rootDirectories.push(window.location.origin + baseUrl);
-for (let moduleName of Object.getOwnPropertyNames(modulePaths)) {
-  let modulePath = modulePaths[moduleName];
-  if (modulePath != moduleName) {
-    customModulePaths[moduleName] = modulePath;
-  }
-  var src = window.location.origin + '/' + modulePath + '.js';
-  if (window.\$dartLoader.moduleIdToUrl.has(moduleName)) {
-    continue;
-  }
-  \$dartLoader.moduleIdToUrl.set(moduleName, src);
-  \$dartLoader.urlToModuleId.set(src, moduleName);
-}
 // Create the main module loaded below.
 define("main_module.bootstrap", ["$entrypoint", "dart_sdk"], function(app, dart_sdk) {
   dart_sdk.dart.setStartAsyncSynchronously(true);
   dart_sdk._debugger.registerDevtoolsFormatter();
-  \$dartLoader.getModuleLibraries = dart_sdk.dart.getModuleLibraries;
+  dart_sdk._isolate_helper.startRootIsolate(() => {}, []);
   if (window.\$dartStackTraceUtility && !window.\$dartStackTraceUtility.ready) {
     window.\$dartStackTraceUtility.ready = true;
     let dart = dart_sdk.dart;
     window.\$dartStackTraceUtility.setSourceMapProvider(
       function(url) {
         url = url.replace(baseUrl, '/');
-        var module = window.\$dartLoader.urlToModuleId.get(url);
-        if (!module) {
-          if (url.endsWith('dart_sdk.js')) {
-            module = 'dart_sdk';
-          } else {
-            try {
-              module = '/packages' + url.split('packages')[1].split('.')[0] + '.dart';
-            } catch (e) {
-              return null;
-            }
-          }
+        var module = window.\$requireLoader.urlToModuleId.get(url)
```

Comment:
It looks like the dartLoader doesn't exist anymore - should I be using the requireLoader? Otherwise  I'm not sure exactly what this code should look like. I tried a few permutations but they all crash

Reply:
You shouldn't need either. You are correct that the `dartLoader` was removed and replaced with `reqruireLoader`. However, `package:dwds` now embeds the `requrieLoader` in the entry bootstrap. 

See here:
https://github.com/dart-lang/webdev/blob/master/dwds/lib/src/handlers/injected_handler.dart#L74

And here:
https://github.com/dart-lang/webdev/blob/master/dwds/lib/src/loaders/require.dart#L179

Let me know if removing this doesn't fix your problem.

Reply:
Removing SGTM, but I don't see any handling of the `dartStackTraceUtility` in the new dwds code

Reply:
@jakemac53 said that logic was safe to remove. He can probably provide more context.

Reply:
Another difference is that If I throw an unhandled exception, before the stackTraceMapper would first be called with a url like `http://localhost:53973/dart_sdk.js` that would be mapped to the module name `dart_sdk`. Now it's failing with `http://localhost:53973/something/dart_stack_trace_mapper.js` which might indicate something isn't getting set up correctly. Still investigating

Reply:
The `dartStackTraceUtility` does still need to exist - it is used for mapping stack traces in the browser console.

Reply:
In this case, it sounds like I should be handling the initialization of `dartStackTraceUtility` and using the require loaded to get  the correct source map?

Reply:
That sounds about right. @jakemac53 should we include this in `package:dwds`?

Reply:
I think it needs to be added by build_web_compilers - it has its own version. Flutter also i think has its own version.

Reply:
Is there an equivalent to this line I need to do for the requireLoader?

https://github.com/dart-lang/build/blob/8c8ed2b52357b62971889ab2c8c26db2163c0080/build_web_compilers/lib/src/dev_compiler_bootstrap.dart#L366

I'm currrently using the following code:

```
  dart_sdk._isolate_helper.startRootIsolate(() => {}, []);
  if (window.\$dartStackTraceUtility && !window.\$dartStackTraceUtility.ready) {
    window.\$dartStackTraceUtility.ready = true;
    let dart = dart_sdk.dart;
    window.\$dartStackTraceUtility.setSourceMapProvider(function(url) {
      url = url.replace(window.\$dartUriBase, '');
      var module = window.\$requireLoader.urlToModuleId.get(url);
      if (!module) return;
      return dart.getSourceMap(url);
    });
  }
```

But I'm only get `url` from the stack trace mapper itself, which leads me to believe that I set it up incorrectly.

Reply:
There is not but I'm happy to add it to `package:dwds` somewhere here: https://github.com/dart-lang/webdev/blob/master/dwds/lib/src/loaders/require.dart#L222

Reply:
Its not so much what API goes into dwds, there is a slight bug now since we use the stack trace mapper from the SDK. This expects there to be a field called `window.$dartLoader.rootDirectories`:

https://github.com/dart-lang/sdk/blob/master/pkg/dev_compiler/web/stack_trace_mapper.dart#L39

All stack traces actually are crashing in the stack trace mapper now. I can work around by defining this in the bootstrap script myself

Reply:
Ahh I see. I'm thinking we just duplicate `window.$dratLoader.rootDirectories` in the DWDS bootstrap for now and eventually clean up the SDK.